entitymanager.js

```diff
@@ -0,0 +1,44 @@
+function EntityManager() {
+  var entities = {};
+  var nextId = 0;
+
+  this.add = function(entity) {
+    entity.registerId(nextId);
+    entities[nextId++] = entity;
+  }
+
+  this.update = function() {
+    for (var index in entities) {
+      if (entities[index].update()) {
+        // Deletes the index property from the entities object.
+        delete entities[index];
+      }
+    }
+  }
+
+  this.render = function() {
+    for (var index in entities) {
+      entities[index].render();
+    }
+  }
+
+  this.checkCollisions = function() {
+    var x = 1;
+    for (var i in entities) {
```

Comment:
This would probably be better expressed with an old-fashioned loop:

    for(var i = 0; i < entities.length; i++) {
      for(var j = i + 1; j < entities.length; j++) {